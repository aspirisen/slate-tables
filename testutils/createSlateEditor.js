import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'
import { Editor } from 'slate-react'

export default function createSlateEditor(props, value) {
	const renderer = new ShallowRenderer()
	renderer.render(React.createElement(Editor, { onChange: () => {}, value, ...props }, null))
	const editor = renderer.getRenderOutput().props.editor
	return editor
}
