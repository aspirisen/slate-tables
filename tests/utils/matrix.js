import fixtures from 'testutils/fixtures'
import createSlateEditor from 'testutils/createSlateEditor'
import Plugin from 'lib/index'

const formatMatrix = matrix => {
	return matrix.map(row => row.map(cell => [cell.virtual, cell.ref.key]))
}

fixtures(__dirname, './fixtures', ({ module }) => {
	const { props, value } = module
	if (!value) {
		throw new Error('No value defined')
	}
	const editor = createSlateEditor({ plugins: [Plugin()], ...props }, value)
	const table = editor.getTableAtKey(editor.value.document.nodes.first().key)
	expect(formatMatrix(table.matrix)).toMatchSnapshot()
})
