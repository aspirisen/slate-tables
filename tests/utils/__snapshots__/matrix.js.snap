// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fixtures matrix-2x3 1`] = `
Immutable.List [
  Immutable.List [
    Array [
      false,
      "2",
    ],
    Array [
      false,
      "4",
    ],
    Array [
      false,
      "6",
    ],
  ],
  Immutable.List [
    Array [
      false,
      "10",
    ],
    Array [
      false,
      "12",
    ],
    Array [
      false,
      "14",
    ],
  ],
]
`;

exports[`fixtures matrix-colspan-1 1`] = `
Immutable.List [
  Immutable.List [
    Array [
      false,
      "1",
    ],
    Array [
      true,
      "1",
    ],
    Array [
      false,
      "3",
    ],
  ],
  Immutable.List [
    Array [
      false,
      "7",
    ],
    Array [
      false,
      "9",
    ],
    Array [
      false,
      "11",
    ],
  ],
]
`;

exports[`fixtures matrix-colspan-2 1`] = `
Immutable.List [
  Immutable.List [
    Array [
      false,
      "2",
    ],
    Array [
      true,
      "2",
    ],
    Array [
      true,
      "2",
    ],
  ],
  Immutable.List [
    Array [
      false,
      "6",
    ],
    Array [
      false,
      "8",
    ],
    Array [
      false,
      "10",
    ],
  ],
]
`;

exports[`fixtures matrix-colspan-3 1`] = `
Immutable.List [
  Immutable.List [
    Array [
      false,
      "2",
    ],
    Array [
      true,
      "2",
    ],
    Array [
      true,
      "2",
    ],
    Array [
      true,
      "2",
    ],
  ],
  Immutable.List [
    Array [
      false,
      "6",
    ],
    Array [
      false,
      "8",
    ],
    Array [
      false,
      "10",
    ],
    Array [
      false,
      "17",
    ],
  ],
]
`;

exports[`fixtures matrix-colspan-4 1`] = `
Immutable.List [
  Immutable.List [
    Array [
      false,
      "2",
    ],
    Array [
      false,
      "5",
    ],
    Array [
      true,
      "5",
    ],
    Array [
      true,
      "5",
    ],
    Array [
      false,
      "26",
    ],
  ],
  Immutable.List [
    Array [
      false,
      "9",
    ],
    Array [
      false,
      "11",
    ],
    Array [
      false,
      "13",
    ],
    Array [
      false,
      "16",
    ],
    Array [
      false,
      "19",
    ],
  ],
]
`;

exports[`fixtures matrix-multi-colspan 1`] = `
Immutable.List [
  Immutable.List [
    Array [
      false,
      "2",
    ],
    Array [
      true,
      "2",
    ],
    Array [
      true,
      "2",
    ],
    Array [
      true,
      "2",
    ],
    Array [
      false,
      "5",
    ],
    Array [
      true,
      "5",
    ],
  ],
]
`;

exports[`fixtures matrix-rowspan 1`] = `
Immutable.List [
  Immutable.List [
    Array [
      false,
      "2",
    ],
    Array [
      false,
      "4",
    ],
  ],
  Immutable.List [
    Array [
      true,
      "2",
    ],
    Array [
      false,
      "already_there",
    ],
  ],
]
`;

exports[`fixtures matrix-simple-1x1 1`] = `
Immutable.List [
  Immutable.List [
    Array [
      false,
      "2",
    ],
    Array [
      false,
      "4",
    ],
  ],
]
`;
